### PR TITLE
PKF-682 Update log read index functionality to work for index lower t…

### PR DESCRIPTION
**Pull request by pradyumnapax**
…han last log read index

K4 mobile app can ask for log read index lower than last read log index. log read functionality used to assume that any log read index is always greater than last read index.

### CODE REVIEW

**Comment by pradyumnapax** [comment]
Test case:

Add few logs to device. One way to do this by connecting to blueteeth K4 mobile app and change intensity, setpoint, lock/unlock.

Disconnect and connect to LightBlue mobile app.
In the app : goto UUID: 64F50310-EFDC-11E6-BC64-92361F002671 and enable 'Listen to notifications'.

In device run cmd :
Should see logs in the mobile device

File: sys/log/full/src/log_fcb.c [hunk 1]
```diff
@@ -399,14 +399,40 @@ log_fcb_walk(struct log *log, log_walk_func_t walk_func,
...
         {
-            return rc;
+            //handle case if we are at end of logs and asked index was somewhere from  0 to n, n being last index
+            memset(&loc, 0, sizeof(loc));
+            rc = fcb_getnext(fcb, &loc);
+            if (rc != 0)
+            {
+                return rc;
+            }
+            rc = log_read_hdr(log, dptr, &ueh);
```
Comment (minchingho-pax) [inline sys/log/full/src/log_fcb.c]: minor: Why do we need a "dptr" for &loc, but we don't need a "somePtr" for &ueh?

Comment (pradyumnapax) [reply sys/log/full/src/log_fcb.c]: good q, maybe can remove 'dptr' variable and directly pass loc, assuming this is what you are asking about.

Comment (pradyumnapax) [reply sys/log/full/src/log_fcb.c]: @minchingho-pax pushed new commit with review comment changes.

**Pull request merged**